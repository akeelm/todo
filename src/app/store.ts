import { configureStore, ThunkAction, Action } from '@reduxjs/toolkit';
import storage from 'redux-persist/lib/storage';
import todoReducer from '../redux/todoSlice';
import { persistReducer } from 'redux-persist'

const persistConfig = {
  key: 'root',
  storage
}

export const store = configureStore({
  reducer: {
    todo: persistReducer(persistConfig, todoReducer),
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
