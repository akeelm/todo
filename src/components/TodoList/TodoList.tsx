import React from 'react';
import ITodo from '../../interfaces/ITodo';
import TodoItem from '../TodoItem';

interface ITodoListProps {
    todoList: ITodo[];
    isCompletedList: boolean;
}

const TodoList: React.FC<ITodoListProps> = ({
    todoList,
    isCompletedList
}) => {
    const completeHandlerFn = (id: number, completed: boolean) => {};
    const deleteHandlerFn = (id: number) => {};
    const editHandlerFn = (id: number, text: string) => {};
    return (
        <>
            <h2>{isCompletedList ? "Todo list" : "Completed list"}</h2>
            <ul
                className="todo-list"
                data-testid="todo-list"
            >
                {
                    todoList.map((x) => 
                        <TodoItem
                            key={`todo-item-${x.id}`}
                            {...x}
                            completeHandler={completeHandlerFn}
                            deleteHandler={deleteHandlerFn}
                            editHandler={editHandlerFn}
                        />
                    )
                }
                {
                    !isCompletedList && (
                        <>
                            <h3>Add a new Todo:</h3>
                            <TodoItem
                                id={-1}
                                text=""
                                completed={false}
                                completeHandler={completeHandlerFn}
                                deleteHandler={deleteHandlerFn}
                                editHandler={editHandlerFn}
                            />
                        </>
                    )
                }
            </ul>
        </>
    );
};

export default TodoList;
