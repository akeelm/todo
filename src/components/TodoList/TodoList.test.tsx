import { render } from '@testing-library/react'
import { Provider } from 'react-redux';
import TodoList from '.';
import { store } from '../../app/store';
import ITodo from '../../interfaces/ITodo';

describe('TodoList tests', () => {
    const mockTodoList: ITodo[] = [
        { id: 1, text: 'Todo 1', completed: false },
        { id: 2, text: 'Todo 2', completed: false },
        { id: 3, text: 'Todo 3', completed: false },
    ];

    it('should be able to render', () => {
       const { container } = render(
           <Provider store={store}>
               <TodoList
                   todoList={mockTodoList}
                   isCompletedList={false}
               />
           </Provider>
       );

       expect(container).toMatchSnapshot();
    });
});