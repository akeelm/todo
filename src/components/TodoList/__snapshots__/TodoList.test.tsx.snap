// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList tests should be able to render 1`] = `
<div>
  <h2>
    Completed list
  </h2>
  <ul
    class="todo-list"
    data-testid="todo-list"
  >
    <li
      class="todo-item"
      data-testid="todo-item"
    >
      <input
        data-testid="todo-item-checkbox"
        tabindex="1"
        title="Complete"
        type="checkbox"
      />
      <span
        class="todo-item__text "
        data-testid="todo-item-text"
        tabindex="1"
      >
        Todo 1
      </span>
      <button
        data-testid="delete-todo-button"
        title="Delete todo"
      >
        Delete
      </button>
    </li>
    <li
      class="todo-item"
      data-testid="todo-item"
    >
      <input
        data-testid="todo-item-checkbox"
        tabindex="1"
        title="Complete"
        type="checkbox"
      />
      <span
        class="todo-item__text "
        data-testid="todo-item-text"
        tabindex="1"
      >
        Todo 2
      </span>
      <button
        data-testid="delete-todo-button"
        title="Delete todo"
      >
        Delete
      </button>
    </li>
    <li
      class="todo-item"
      data-testid="todo-item"
    >
      <input
        data-testid="todo-item-checkbox"
        tabindex="1"
        title="Complete"
        type="checkbox"
      />
      <span
        class="todo-item__text "
        data-testid="todo-item-text"
        tabindex="1"
      >
        Todo 3
      </span>
      <button
        data-testid="delete-todo-button"
        title="Delete todo"
      >
        Delete
      </button>
    </li>
    <h3>
      Add a new Todo:
    </h3>
    <li
      class="todo-item"
      data-testid="todo-item"
    >
      <input
        data-testid="todo-item-checkbox"
        tabindex="1"
        title="Complete"
        type="checkbox"
      />
      <span
        class="todo-item__text "
        data-testid="todo-item-text"
        tabindex="1"
      />
      <button
        data-testid="delete-todo-button"
        title="Delete todo"
      >
        Delete
      </button>
    </li>
  </ul>
</div>
`;
